physical/raft/chunking_test.go

```diff
@@ -176,3 +181,47 @@ func TestFSM_Chunking_TermChange(t *testing.T) {
 		}
 	}
 }
+
+func TestRaft_Chunking_AppliedIndex(t *testing.T) {
+	t.Parallel()
+
+	raft, dir := getRaft(t, true, false)
+	defer os.RemoveAll(dir)
+
+	// Lower the size for tests
+	raftchunking.ChunkSize = 1024
+	val, err := uuid.GenerateRandomBytes(3 * raftchunking.ChunkSize)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	currentIndex := raft.AppliedIndex()
+	// Write some data
+	for i := 0; i < 10; i++ {
+		err := raft.Put(context.Background(), &physical.Entry{
+			Key:   fmt.Sprintf("key-%d", i),
+			Value: val,
```

Comment:
Consider incorporating (e.g. as a prefix) `i` to the stored value, and checking it below, to strengthen the test against entry data being associated with the wrong key.